src/fromager/sources.py

```diff
@@ -605,8 +608,19 @@ def build_sdist(
             build_dir=build_dir,
             build_env=build_env,
         )
-    logger.info(f"built source distribution {sdist_filename}")
-    return sdist_filename
+    logger.info(f"built source distribution {sdist_file}")
+
+    # validate location and file name
+    if not sdist_file.is_file():
+        raise FileNotFoundError(sdist_file)
+    sdist_file = sdist_file.resolve()
+    if sdist_file.parent != ctx.sdists_builds:
+        raise ValueError(
+            "{sdist_file!} is not in sdists build directory {ctx.sdists_builds!r}"
```

Comment:
We need file formatting for this error. `f"{sdist_file!r} is not in sdists build directory {ctx.sdists_builds!r}"`

Reply:
Good catch!